stripe/functions/index.js

```diff
@@ -63,6 +63,22 @@ exports.createStripeCustomer = functions.auth.user().onCreate(event => {
   });
 });
 
+// Add a payment source (card) for a user by writing a stripe payment source token to Realtime database
+exports.addPaymentSource = functions.database.ref('/stripe_customers/{userId}/cards/{token}').onWrite(event => {
+  const token = event.data.val();
+  const user = event.params.userId;
+  return stripe.customers.createSource(
+    user,
+    {source: token}
+  ).then(response => {
+        return console.log('payment source added successfully for user', user);
```

Comment:
what if instead we wrote back the response `return event.data.ref.parent.set(response)`, that's what I am doing in my scratch project then you have VISA ***1234, exp date, etc. to allow selection